File: nodejs/checkio/src/initiation/replace_first.ts
import assert from "assert";

function replaceFirst(values: number[]): number[] {
	return values.length > 0 ? [...values.slice(1), values[0]] : [];
}

console.log("Example:");
console.log(JSON.stringify(replaceFirst([1, 2, 3, 4])));

// These "asserts" are used for self-checking
assert.deepStrictEqual(replaceFirst([1, 2, 3, 4]), [2, 3, 4, 1]);
assert.deepStrictEqual(replaceFirst([1]), [1]);
assert.deepStrictEqual(replaceFirst([]), []);

console.log("Coding complete? Click 'Check Solution' to earn rewards!");

File: python/checkio/initiation/replace-first.py
from collections.abc import Iterable


def replace_first(items: list) -> Iterable:
    return items[1:] + items[:1]


# These "asserts" are used for self-checking
print("Example:")
print(list(replace_first([1, 2, 3, 4])))

assert replace_first([1, 2, 3, 4]) == [2, 3, 4, 1]
assert replace_first([1]) == [1]
assert replace_first([]) == []

print("The mission is done! Click 'Check Solution' to earn rewards!")
